ports/assaultshell/ASSAULT SHELL.sh

```diff
@@ -0,0 +1,45 @@
+#!/bin/bash
+
+XDG_DATA_HOME=${XDG_DATA_HOME:-$HOME/.local/share}
+
+if [ -d "/opt/system/Tools/PortMaster/" ]; then
+  controlfolder="/opt/system/Tools/PortMaster"
+elif [ -d "/opt/tools/PortMaster/" ]; then
+  controlfolder="/opt/tools/PortMaster"
+elif [ -d "$XDG_DATA_HOME/PortMaster/" ]; then
+  controlfolder="$XDG_DATA_HOME/PortMaster"
+else
+  controlfolder="/roms/ports/PortMaster"
+fi
+
+source $controlfolder/control.txt
+source $controlfolder/device_info.txt
+export PORT_32BIT="Y"
+[ -f "${controlfolder}/mod_${CFW_NAME}.txt" ] && source "${controlfolder}/mod_${CFW_NAME}.txt"
+get_controls
+
+$ESUDO chmod 666 /dev/tty0
+
+GAMEDIR="/$directory/ports/assaultshell"
+
+export LD_LIBRARY_PATH="/usr/lib32:$GAMEDIR/libs:$GAMEDIR/utils/libs":$LD_LIBRARY_PATH
+export GMLOADER_DEPTH_DISABLE=1
+export GMLOADER_SAVEDIR="$GAMEDIR/gamedata/"
+export GMLOADER_PLATFORM="os_linux"
+
+cd "$GAMEDIR"
+> "$GAMEDIR/log.txt" && exec > >(tee "$GAMEDIR/log.txt") 2>&1
+
+[ -f "./gamedata/data.win" ] && mv gamedata/data.win gamedata/game.droid
+
+$GPTOKEYB "gmloader" &
+echo "Loading, please wait... " > /dev/tty0
+
+$ESUDO chmod +x "$GAMEDIR/gmloader"
+
+./gmloader assaultshell.apk
+
+$ESUDO kill -9 "$(pidof gptokeyb)"
```

Comment:
The rest of the script can be replaced with this:

```suggestion
pm_finish
```